File: loan-repayment-java/src/main/java/com/loan/repayment/entity/BorrowerPayment.java
package com.loan.repayment.entity;

import java.time.ZonedDateTime;

import io.swagger.annotations.ApiModelProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * 
 * @author Ramachandra
 *
 */
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Getter
@EqualsAndHashCode
public class BorrowerPayment {

    /**
     * The annuity payment amount.
     */
    @ApiModelProperty(notes = "The annuity payment amount", example = "0.48")
    private String borrowerPaymentAmount;

    /**
     * The loan payment date.
     */
    @ApiModelProperty(notes = "The loan payment date")
    private ZonedDateTime date;

    /**
     * Initial Outstanding Principal pending for the current month.
     */
    @ApiModelProperty(notes = "Initial Outstanding Principal pending " + "for the current month", example = "100.0")
    private String initialOutstandingPrincipal;

    /**
     * Interest for the month.
     */
    @ApiModelProperty(notes = "Interest for the month", example = "0.42")
    private String interest;

    /**
     * Principal for the month for the month.
     */
    @ApiModelProperty(notes = "Principal for the month", example = "0.06")
    private String principal;

    /**
     * Outstanding Principal after the interset paid for the month.
     */
    @ApiModelProperty(notes = "Outstanding Principal after the interset " + "paid for the month", example = "99.94")
    private String remainingOutstandingPrincipal;

}

File: loan-repayment-kotlin/src/main/kotlin/com/loan/repayment/entity/BorrowerPayment.kt
package com.loan.repayment.entity

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.annotations.ApiModelProperty
import java.time.ZonedDateTime

/**
 *
 * @author Rama chandra
 */
data class BorrowerPayment(
        @ApiModelProperty(notes = "The annuity payment amount" , example = "0.48")
        @JsonProperty("borrowerPaymentAmount")
        val borrowerPaymentAmount : String ,
        @ApiModelProperty(notes = "The loan payment date")
        @JsonProperty("date")
        val date : ZonedDateTime ,
        @ApiModelProperty(notes = "Initial Outstanding Principal pending " + "for the current month" , example = "100.0")
        @JsonProperty("initialOutstandingPrincipal")
        val initialOutstandingPrincipal : String ,
        @ApiModelProperty(notes = "Interest for the month" , example = "0.42")
        @JsonProperty("interest")
        val interest : String ,
        @ApiModelProperty(notes = "Principal for the month" , example = "0.06")
        @JsonProperty("principal")
        val principal : String ,
        @ApiModelProperty(notes = "Outstanding Principal after the interest " + "paid for the month" , example = "99.94")
        @JsonProperty("remainingOutstandingPrincipal")
        val remainingOutstandingPrincipal : String
)